raiden-ts/src/transport/epics.ts

```diff
@@ -1503,3 +1520,352 @@ export const deliveredEpic = (
+) {
+  const { callId, peerId, peerAddress } = info;
+  const { log, latest$, config$ } = deps;
+
+  const connection = new RTCPeerConnection({
+    iceServers: [{ urls: 'stun:stun.l.google.com:19302' }],
+  });
+  // we relay on retries, no need to enforce ordered
+  const channel = connection.createDataChannel(callId, { ordered: false });
+  return merge(
+    // despite 'never' emitting, candidates$ have side-effects while/when subscribed
+    handleCandidates$(connection, matrix, start$, info, deps),
+    defer(() => connection.createOffer()).pipe(
+      mergeMap((offer) => {
+        connection.setLocalDescription(offer);
+        const content = {
+          call_id: callId,
+          lifetime: httpTimeout,
+          version: 0,
+          offer,
+        };
+        return merge(
+          // wait for answer
+          fromEvent<MatrixEvent>(matrix, 'event').pipe(
+            filter<MatrixEvent>(
+              // FIXME: remove any when MatrixEvent type exposes getAge & getContent methods
+              // eslint-disable-next-line @typescript-eslint/no-explicit-any
+              (event: any) =>
+                event.getType() === ('m.call.answer' as EventType) &&
+                event.getSender() === peerId &&
+                event.getContent()?.call_id === callId &&
+                event.getAge() <= (event.getContent()?.lifetime ?? httpTimeout) &&
+                !!event.getContent()?.answer,
+            ),
+          ),
+          // send invite with offer
+          waitMemberAndSend$(peerAddress, matrix, 'm.call.invite' as EventType, content, {
+            log,
+            latest$,
+            config$,
+          }).pipe(
+            tap((e) => log.debug('RTC: sent invite', callId, e)),
+            ignoreElements(),
+          ),
+        );
+      }),
+      take(1),
+      tap(() => log.info('RTC: got answer', callId)),
+      map((event) => {
+        connection.setRemoteDescription(new RTCSessionDescription(event.event.content.answer));
+        start$.next(null);
+        start$.complete();
+      }),
+      ignoreElements(),
+    ),
+    of(channel), // output created channel
+  );
+}
+
+// setup channel for callee
+function setupCalleeChannel$(
+  matrix: MatrixClient,
+  start$: Subject<null>,
+  info: CallInfo,
+  { httpTimeout }: RaidenConfig,
+  deps: Pick<RaidenEpicDeps, 'log' | 'latest$' | 'config$'>,
+) {
+  const { callId, peerId, peerAddress } = info;
+  const { log, latest$, config$ } = deps;
+  return fromEvent<MatrixEvent>(matrix, 'event').pipe(
+    filter(
+      // FIXME: remove any when MatrixEvent type exposes getAge & getContent methods
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (event: any) =>
+        event.getType() === ('m.call.invite' as EventType) &&
+        event.getSender() === peerId &&
+        event.getContent()?.call_id === callId &&
+        event.getAge() <= (event.getContent()?.lifetime ?? httpTimeout) &&
+        !!event.getContent()?.offer,
```

Comment:
nitpick: this filter seems really similar to the other own for the answer, any chance we could some how extract the common part?

Reply:
Done, nice catch.. the function got a little bit big, because of the stricter typing inference on the returned event, but worked for all 4 cases (events), thanks for the suggestion